crates/gossip/src/gossip.rs

```diff
@@ -250,6 +251,18 @@ impl K2Gossip {
             target_peer_url
         );
 
+        // Before we send the initiate message, check whether the target has already
```

Comment:
I was thinking that checking this as late as possible would minimize the race between two peers trying to initiate with each other. Was this causing a problem?

Reply:
I forgot to comment here. It wasn't causing a problem, no. Just seemed negligible the operation to look up local agents.

Reply:
Undone